src/raw/bson.rs

```diff
@@ -244,245 +253,44 @@ impl<'a> RawBson<'a> {
             _ => None,
         }
     }
-}
-
-/// A visitor used to deserialize types backed by raw BSON.
-pub(crate) struct RawBsonVisitor;
```

Comment:
This was moved to `src/raw/serde.rs` and slightly modified so that it could be shared with `OwnedRawBson`.